#scripts/update-mode-asm/index.js

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env node
+
+const fs = require('fs');
+const path = require('path');
+const regexCombiner = require('regex-combiner');
+
+// read list file as array of lines
+const listContents = fs.readFileSync(path.resolve(__dirname, './asm-instructions.txt'))
+                     .toString().split('\n');
+
+const combined = regexCombiner(listContents);
+const regexPattern = combined.toString();
+
+const modAsmFile = path.resolve(__dirname, '../../source/WebApp/components/internal/codemirror/mode-asm.ts');
+let asmScriptContents = fs.readFileSync(modAsmFile).toString();
+
+console.log('size before: ', fs.statSync(modAsmFile).size, ' bytes');
+
+asmScriptContents = asmScriptContents.replace(/^const generatedRegex =.*$/mg, `const generatedRegex = new RegExp(${regexPattern});`);
+
+fs.writeFileSync(modAsmFile, asmScriptContents);
+
+console.log('size after: ', fs.statSync(modAsmFile).size, ' bytes');
```

Comment:
This can probably be a build task in `WebApp/scripts.ts`. Can even be simplified to something like this:
```
codemirror/mode-asm.ts
codemirror/mode-asm-instructions.txt
codemirror/mode-asm-instructions.ts // generated from mode-asm-instructions.txt
```

Then you can do `import instructionRegex from './mode-asm-instructions';` in `codemirror/mode-asm.ts` with need for search/replace the `mode-asm.ts` code.

Reply:
(I don't mind moving it, just letting you know in case you want to)

Reply:
Does npm build task run in the CI? Locally, with tip of edge branch (without any changes) I get:

```sh
$ cd scripts/WebApp
$ npm i
$ npm run-script build

> sharplab@0.0.0 build /Users/am11/projects/SharpLab/source/WebApp
> ts-node-script ./scripts.ts build

task build starting...
task less starting...
task ts starting...
task ts-lint starting...
task icons starting...
task manifest starting...
task html starting...
task latest starting...
task manifest failed:
TypeError: content.icons.flatMap is not a function
    at oldowan_1.task (/Users/am11/projects/SharpLab/source/WebApp/scripts.ts:103:35)
task build failed: task manifest failed
```

Reply:
Does npm build task run in the CI? Locally, with tip of edge branch (without any changes) I get:

```sh
$ cd source/WebApp
$ npm i
$ npm run-script build

> sharplab@0.0.0 build /Users/am11/projects/SharpLab/source/WebApp
> ts-node-script ./scripts.ts build

task build starting...
task less starting...
task ts starting...
task ts-lint starting...
task icons starting...
task manifest starting...
task html starting...
task latest starting...
task manifest failed:
TypeError: content.icons.flatMap is not a function
    at oldowan_1.task (/Users/am11/projects/SharpLab/source/WebApp/scripts.ts:103:35)
task build failed: task manifest failed
```

Reply:
Updated as a task (called it `asm` next to `html`). CI gets green but locally I am seeing that `content.icons.flatMap` error from the linter (commented it out to successfully run the tasks and then later uncommented like this: https://github.com/ashmind/SharpLab/pull/586/commits/52959d21f3a7c42a6d2845998388b8e60f4ee28a).

Reply:
Ah, my node was v10, flatMap was introduced in v11, all green after the upgrade to v14. Seems to be running in GitHub Actions CI (but not AppVeyor).

Reply:
Thanks -- we can ignore AppVeyor, I just moved JS build out of there anyways.